man/ncid-wakeup.1
.\" %W% %G%
.TH NCID-WAKEUP 1
.SH NAME
.B ncid-wakeup
.SH SYNOPSIS
ncid --wakeup
.B ncid-wakeup
.SH DESCRIPTION
The
.B ncid-wakeup
output module is called to wakeup X-Windows for ncid in GUI mode
or ncid using a output module.  If another module is called, it
is executed before that module.  Since no information regarding
the call is required, none is sent.
It is called when the WakeUp option is set either on the command
line of in the \fIncid.conf\fR file.
.SH REQUIREMENTS
This module only works with X-windows and Gnome.
.br
Do not set the ncid "--program | -P" option.
.SH FILES
/etc/ncid/ncid.conf
.br
/usr/share/ncid/ncid-skel
.SH SEE ALSO
ncidd.8,
ncid-modules.7,
ncid.1,
ncid-alert.1,
ncid-initmodem.1,
ncid-kpopup.1,
ncid-mythtv.1,
ncid-notify.1,
ncid-page.1,
ncid-samba.1,
ncid-skel.1,
ncid-speak.1,
ncid-yac.1


doc/CHANGES
ncid - Network CallerID

Version 0.87
    - README: updated
    - Makefile: added test: $(DOCDIR) does not exist, create it
    - All GPL license headers in files were updated to GPL 3
    - client/ncid.sh: logClock{} changed {^(CID|HUP|OUT)} to {^(?:CID|HUP|OUT)}
                      added "Line Labels" in help menu and labInfo {}
                      checkTypes() added new types and changed type numbers
                      getCID(): added new types and changed if statements
                                for new Type numbers
    - debian/Makefile: updated
    - debian/README.Debian: updated
    - debian/*.init: removed "S" from default STOP
    - debian/wc2ncid.init: removed configuration stuff, changed sbin to sbin
    - debian/yac2ncid.init: removed configuration stuff, changed sbin to sbin
    - debian/rn2ncid.init: new
    - doc/INSTALL-Ubuntu: updated
    - doc/INSTALL-Raspbian: updated
    - doc/LICENSE: updated to GPL 3
    - doc/Setup-rn2ncid: new
    - doc/Verbose-rn2ncid: new
    - Fedora/Makefile: updated
    - Fedora/README.Fedora: updated
    - Fedora/rn2ncid.init: new
    - Fedora/rn2ncid.service: new
    - FreeBSD/Makefile: updated
    - FreeBSD/README.FreeBSD: updated
    - FreeBSD/rn2ncid.rc: new
    - gateway/Makefile: added $(EXTRA_CFLAGS) to CFLAGS definition
    - gateway/README.Gateways: renamed to README.gateways and updated
    - gateway/*.[ch]: updated license in headers
    - gateway/*.pl: updated license in comments
    - gateway/sip2ncid.h: added UNAUTHORIZED define
    - gateway/sip2ncid.c: processPackets() added UNAUTHORIZED section
    - gateway/wc2ncid.conf: - improved comment wording
                            - changed nciddelay to delay
    - gateway/wc2ncid.pl: - moved test mode just after Command Line Processing
                          - moved logfile and config file messages to
                            after start message
                          - added cleanup(), called by errorExit() & sigHandle()
                          - added delay option to pod documentation wording,
                            improved some pod wording
                          - changed $nciddelay to $delay, added $Delay,
                            added --delay|-d option
    - gateway/rn2ncid.pl: new
    - gateway/rn2ncid.config: new
    - Mac/Makefile: updated
    - Mac/README.Mac: updated
    - Mac/net.sourceforge.rn2ncid.plist: new
    - man/Makefile: updated
    - man/ncid-*.1: updated
    - man/ncid.1: updated
    - man/ncid.conf.5: updated
    - man/ncidd.8: updated
    - man/yac2ncid.1: updated
    - man/rn2ncid.conf.5: new
    - modules/ncid-*.sh: updated
    - modules/ncid-*.conf: updated
    - modules/ncid-notify.sh: changed line one from /bin/sh to /bin/bash
                              minor changes to verbose output and comments
    - modules/ncid-speak.sh: added "AreaCodeLength" code to speak the area
                             code instead or "NO NAME" or entire number
    - modules/ncid-speak.conf-in: added "AreaCodeLength" and comments
    - server/ncidd.c: writeLog() - moved "logMsg(LEVEL3, msgbuf);" to start
                      formatCID() - moved cidnoname test inside
                                    if (!(cid.status & CIDNMBR))
                                 - added PID:
                      added BLKLINE, NOTLINE, PIDLINE to *lineTags[]
                      doPoll: - added code for a NOT: message
    - server/ncidd.h: added defines for PID, PIDLINE, NOTLINE, CALLPID
    - test/README.test: updated
    - test/*: renamed test programs and test data
    - test/test-rn2ncid: new
    - test/test-rn2ncid.data: new
    - TiVo/ncid-tivo.sh:  if no line label, display $CIDNAME/$CIDNMBR
                          else display "$CIDNAME $CIDNMBR/$CIDTYPE $CIDLINE"

Version 0.86.1
    - debian/changelog: updated
    - debian/control: for package ncid-client added "curl" to Depends

Version 0.86
    - README: updated
              added redhat: and redhat-install:
    - attic/*: new
    - client/ncid.sh: added more doVerbose statements and levels
                      changed 300* doVerbose message to any one of four
                      added line count to "End of call log" message
                      removed the --raw | -R option
                      fixed and improved call on a ring number
                      fonts used depend on platform and OS
                      allow resizing of window
                      allow font size changes
                      time format in history window can be either 12 and 24
                      new menu called Preferences for Fonts and time format
                      menu item to save window size
                      menu item to save window size and location
                      added the "-h | --help" option
                      added the "-v | --version" option
                      added the "-w | --wakeup" option
                      getCID{}: added calls to ncid-wakeup module
                      changed "-V" to be the "-v | --verbose" option
                      makeWindow{}: improved widget and menu code
                      added saveSize{} write_rc_file{} logFont{} changeFont{}
                      getField{}: changed regex to include "*" and "-"
    - client/ncid.conf-in: removed MsgFlag variable and comments
                           removed Raw variable and comments
                           added comments to CallOnRing section
                           added comments to Ring Options section
                           new "WakeUp" section
                           new "WrapLines" section
    - debian/Makefile: updated
    - debian/wc2ncid.init: new
    - debian/ncidsip.init: removed
    - doc/CONTRIBUTORS: updated
    - doc/NCID-SDK.odt: updated
    - doc/Setup-ncidsip: removed
    - doc/Setup-sip2ncid: updated
    - doc/Setup-wc2ncid: new
    - doc/INSTALL-Redhat: new
    - doc/INSTALL*: updated
    - doc/INSTALL-RPi: new
    - Fedora/Makefile: updated

modules/ncid-wakeup.sh
#!/bin/sh

# Wakeup Output Module

# Last modified: Tue Jan 22, 2013

# This output module is called to wakeup X-Windows for ncid
# in GUI mode or using a output module.  If another module
# is called, it is executed before that module.  Since no
# information regarding the call is required, none is sent.
#
# IMPORTANT: This module only works with X-windows and Gnome.
#            It is called when the WakeUp option is set.
#            Do not set the ncid "--program | -P" option.
#
# ncid usage:
#   ncid --wakeup
#   ncid (when WakeUP in ncid.conf is set to 1)

#ConfigDir=/usr/local/etc/ncid
#ConfigFile=$ConfigDir/ncidmodules.conf
#
#[ -f $ConfigFile ] && . $ConfigFile

VERSION=$(gnome-screensaver-command  -V | sed 's/[^0-9]*\([0-9]*\).*/\1/')

if [ $VERSION -ge 3 ]
then
    # Reset the display's timeout values
    /usr/bin/xset -display $DISPLAY s reset

    # Deactivate the gnome screensaver.
    /usr/bin/gnome-screensaver-command --deactivate

else
    # Deactivate the screensaver
    /usr/bin/xset -display $DISPLAY dpms force on

    # Simulate a keystroke to reset the display's timeout values
    /usr/bin/gnome-screensaver-command --poke
fi

exit 0
